plenoptic/simulate/models/frontend.py

```diff
@@ -47,63 +47,68 @@ class Front_End(nn.Module):
-                                     dtype=torch.float32).unsqueeze(1))
+        # contrast and luminance normalization scaling
+        self.luminance_scale = nn.Parameter(torch.rand((1, 2, 1, 1), dtype=dtype))
+        self.contrast_scale = nn.Parameter(torch.rand((1, 2, 1, 1), dtype=dtype))
 
-        luminance = nn.Conv2d(in_channels=1, out_channels=2, kernel_size=31,
-                              bias=False)
-        luminance.weight = nn.Parameter(torch.tensor((filts[1], filts[4]),
-                                        dtype=torch.float32).unsqueeze(1))
+        # pad all transforms for convolution
+        pad = nn.ReflectionPad2d(self.linear.weight.shape[-1]//2)
+        self.linear_pad = transforms.Compose([pad, self.linear])
+        self.luminance_pad = transforms.Compose([pad, self.luminance])
+        self.contrast_pad = transforms.Compose([pad, self.contrast])
+        self.softplus = nn.Softplus()
 
-        contrast = nn.Conv2d(in_channels=2, out_channels=2, kernel_size=31,
-                             groups=2, bias=False)
-        contrast.weight = nn.Parameter(torch.tensor((filts[2], filts[5]),
-                                       dtype=torch.float32).unsqueeze(1))
+        self.disk_mask = disk_mask
+        self._disk = None  # cached disk to apply to image
 
-        pad = nn.ReflectionPad2d(filts.shape[-1]//2)
+        if pretrained:
+            self._load_pretrained(dtype)
 
-        self.luminance_scals = torch.tensor((scals[0], scals[2]), dtype=torch.float32)
-        self.luminance_scals = self.luminance_scals.view((1, 2, 1, 1))
+        if not requires_grad:  # turn off gradient
+            [p.requires_grad_(False) for p in self.parameters()]
 
-        self.contrast_scals = torch.tensor((scals[1], scals[3]), dtype=torch.float32)
-        self.contrast_scals = self.luminance_scals.view((1, 2, 1, 1))
+    def _load_pretrained(self, dtype):
+        """Load FrontEnd model weights used from Berardino et al (2017)"""
+        filts = np.load(dirname + '/front_end_filters.npy')
+        scals = np.load(dirname + '/front_end_scalars.npy')
 
-        del filts, scals
+        # conv kernels
+        self.linear.weight.data = torch.tensor((filts[0], filts[3]), dtype=dtype).unsqueeze(1)
+        self.luminance.weight.data = torch.tensor((filts[1], filts[4]), dtype=dtype).unsqueeze(1)
+        self.contrast.weight.data = torch.tensor((filts[2], filts[5]), dtype=dtype).unsqueeze(1)
 
-        self.linear = transforms.Compose([pad, linear])
-        self.luminance = transforms.Compose([pad, luminance])
-        self.contrast = transforms.Compose([pad, contrast])
-        self.softplus = nn.Softplus()
-
-        self.disk_mask = disk_mask
+        # normalization scale factors
+        self.luminance_scale.data = torch.tensor((scals[0], scals[2]), dtype=dtype).view((1, 2, 1, 1))
+        self.contrast_scale.data = torch.tensor((scals[1], scals[3]), dtype=dtype).view((1, 2, 1, 1))
```

Comment:
I think it's because the 1-channel input is split into two channels that each have a linear+nonlinear steps that happen in parallel. So the ordering was wrt the steps within each channel.